contracts/asset/AaveV3Adapter.sol

```diff
@@ -0,0 +1,186 @@
+        address _assetManager,
+        LendingPool3 _lendingPool,
+        AMarket3 _market
+    ) public initializer {
+        Controller.__Controller_init(msg.sender);
+        assetManager = _assetManager;
+        lendingPool = _lendingPool;
+        market = _market;
+    }
+
+    function setAssetManager(address _assetManager) external onlyAdmin {
+        assetManager = _assetManager;
+    }
+
+    function setFloor(address tokenAddress, uint256 floor) external onlyAdmin {
+        require(tokenAddress != address(0), "AaveAdapter: tokenAddress can not be zero");
+        floorMap[tokenAddress] = floor;
+    }
+
+    function setCeiling(address tokenAddress, uint256 ceiling) external onlyAdmin {
+        require(tokenAddress != address(0), "AaveAdapter: tokenAddress can not be zero");
+        ceilingMap[tokenAddress] = ceiling;
+    }
+
+    function mapTokenToAToken(address tokenAddress) external onlyAdmin {
+        address aTokenAddress;
+        (, , , , , , , , aTokenAddress, , , , , , ) = lendingPool.getReserveData(tokenAddress);
+        IERC20Upgradeable token = IERC20Upgradeable(tokenAddress);
+        token.safeApprove(address(lendingPool), 0);
+        token.safeApprove(address(lendingPool), UINT256_MAX);
+        tokenToAToken[tokenAddress] = aTokenAddress;
+    }
+
+    function getRate(address tokenAddress) external view override returns (uint256) {
+        uint128 currentLiquidityRate;
+        (, , currentLiquidityRate, , , , , , , , , , , , ) = lendingPool.getReserveData(tokenAddress);
+        return uint256(currentLiquidityRate);
+    }
+
+    function deposit(address tokenAddress) external override checkTokenSupported(tokenAddress) {
+        IERC20Upgradeable token = IERC20Upgradeable(tokenAddress);
+        uint256 amount = token.balanceOf(address(this));
+        lendingPool.deposit(tokenAddress, amount, address(this), 0);
+    }
+
+    function withdraw(
+        address tokenAddress,
+        address recipient,
+        uint256 tokenAmount
+    ) external override onlyAssetManager checkTokenSupported(tokenAddress) {
+        // slither-disable-next-line unused-return
+        lendingPool.withdraw(tokenAddress, tokenAmount, recipient);
+    }
+
+    function withdrawAll(address tokenAddress, address recipient)
+        external
+        override
+        onlyAssetManager
+        checkTokenSupported(tokenAddress)
+    {
+        // slither-disable-next-line unused-return
+        lendingPool.withdraw(tokenAddress, UINT256_MAX, recipient);
+    }
+
+    function claimTokens(address tokenAddress, address recipient) external override onlyAssetManager {
+        _claimTokens(tokenAddress, recipient);
+    }
+
+    function _getSupply(address tokenAddress) private view returns (uint256) {
+        address aTokenAddress = tokenToAToken[tokenAddress];
+        AToken aToken = AToken(aTokenAddress);
+        uint256 balance = aToken.balanceOf(address(this));
+        if (balance <= 10) {
```

Comment:
Whats the purpose of this check? Assuming to deal with some dust scenario but why would we want to report 0 when we have some dust?